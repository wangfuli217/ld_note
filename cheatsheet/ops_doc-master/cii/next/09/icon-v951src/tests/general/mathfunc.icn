#SRC: JCON

#  test math functions including argument coercion

procedure main()
   every try(sqrt, 0.0 | 0.25 | 1 | 2.25 | "4" | 256.0)
   every try(log, 0.1 | 0.25 | 1.0 | &e | "10" | 100 | 1000)
   every try(log, 0.1 | 0.25 | 1.0 | &e | "10" | 100 | 1000, 10)
   every try(exp, -3 | -1 | 0 | 1.0 | "2" | '3')
   write()
   every try(dtor, -1000 | -360 | -90 | '0' | "45" | 360 | 1000)
   every try(rtod, -3 * &pi | -2 * &pi | -&pi | '0' | &pi / 3 | &pi / 2 |
   	"3.1415936535" | 2 * &pi)
   every try(sin | cos | tan, -&pi / 3 | '0' | "3.1415926535")
   every try(asin | acos, '-1' | -0.5 | "0" | 0.5 | 1.0)
   every try(atan, "-1000" | -47 | -1 | -.5 | '0' | .5 | 1 | 47 | 1000) 
   every try(atan, -1 | '0' | "1", '-1' | "0" | 1)
end


#  try(p, a, b) -- print call to p(a,b) and its results

procedure try(p, a, b)
   write(image(p,15), "(", r(a), (", "||r(\b)) | &null, ") = ", r(p(a,b)))
   return
end


#  r(v) -- round v to have only 3 digits after the decimal point

procedure r(v)
   if v >= 0 then
      return integer(v * 1000 + 0.5) / 1000.0
   else
      return integer(v * 1000 - 0.5) / 1000.0
end
