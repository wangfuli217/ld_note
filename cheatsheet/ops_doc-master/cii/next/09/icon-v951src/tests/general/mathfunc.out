function sqrt(0.0) = 0.0
function sqrt(0.25) = 0.5
function sqrt(1.0) = 1.0
function sqrt(2.25) = 1.5
function sqrt(4.0) = 2.0
function sqrt(256.0) = 16.0
function log(0.1) = -2.303
function log(0.25) = -1.386
function log(1.0) = 0.0
function log(2.718) = 1.0
function log(10.0) = 2.303
function log(100.0) = 4.605
function log(1000.0) = 6.908
function log(0.1, 10.0) = -1.0
function log(0.25, 10.0) = -0.602
function log(1.0, 10.0) = 0.0
function log(2.718, 10.0) = 0.434
function log(10.0, 10.0) = 1.0
function log(100.0, 10.0) = 2.0
function log(1000.0, 10.0) = 3.0
function exp(-3.0) = 0.05
function exp(-1.0) = 0.368
function exp(0.0) = 1.0
function exp(1.0) = 2.718
function exp(2.0) = 7.389
function exp(3.0) = 20.086

function dtor(-1000.0) = -17.453
function dtor(-360.0) = -6.283
function dtor(-90.0) = -1.571
function dtor(0.0) = 0.0
function dtor(45.0) = 0.785
function dtor(360.0) = 6.283
function dtor(1000.0) = 17.453
function rtod(-9.425) = -540.0
function rtod(-6.283) = -360.0
function rtod(-3.142) = -180.0
function rtod(0.0) = 0.0
function rtod(1.047) = 60.0
function rtod(1.571) = 90.0
function rtod(3.142) = 180.0
function rtod(6.283) = 360.0
function sin(-1.047) = -0.866
function sin(0.0) = 0.0
function sin(3.142) = 0.0
function cos(-1.047) = 0.5
function cos(0.0) = 1.0
function cos(3.142) = -1.0
function tan(-1.047) = -1.732
function tan(0.0) = 0.0
function tan(3.142) = 0.0
function asin(-1.0) = -1.571
function asin(-0.5) = -0.524
function asin(0.0) = 0.0
function asin(0.5) = 0.524
function asin(1.0) = 1.571
function acos(-1.0) = 3.142
function acos(-0.5) = 2.094
function acos(0.0) = 1.571
function acos(0.5) = 1.047
function acos(1.0) = 0.0
function atan(-1000.0) = -1.57
function atan(-47.0) = -1.55
function atan(-1.0) = -0.785
function atan(-0.5) = -0.464
function atan(0.0) = 0.0
function atan(0.5) = 0.464
function atan(1.0) = 0.785
function atan(47.0) = 1.55
function atan(1000.0) = 1.57
function atan(-1.0, -1.0) = -2.356
function atan(-1.0, 0.0) = -1.571
function atan(-1.0, 1.0) = -0.785
function atan(0.0, -1.0) = 3.142
function atan(0.0, 0.0) = 0.0
function atan(0.0, 1.0) = 0.0
function atan(1.0, -1.0) = 2.356
function atan(1.0, 0.0) = 1.571
function atan(1.0, 1.0) = 0.785
